File: 1931_회의실/solution.py
def solve(_meetings):
    end_time = 0
    cnt = 0

    for j in range(len(_meetings)):
        meeting = _meetings[j]
        if end_time <= meeting[0]:
            end_time = meeting[1]
            cnt += 1

    return cnt


meetings = []
total_meeting_cnt = int(input())

for i in range(total_meeting_cnt):
    start, end = map(int, input().split())
    meetings.append([start, end])

meetings.sort(key=lambda x: (x[1], x[0]))

print(solve(meetings))

File: 1931_회의실/solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

struct Time {
    int begin;
    int end;
};

bool cmp(Time a, Time b) {
    if (a.end == b.end) {
        return a.begin < b.begin;
    } else {
        return a.end < b.end;
    }
}

int solve(vector<Time>& t) {
    int cnt = 0;
    int end_time = 0;
    for (int i = 0; i < t.size(); i++) {
        if (end_time <= t[i].begin) {
            end_time = t[i].end;
            cnt++;
        }
    }
    return cnt;
}

int main() {
    int N;
    cin >> N;

    vector<Time> t(N);

    for (int i = 0; i < N; i++) {
        cin >> t[i].begin >> t[i].end;
    }

    sort(t.begin(), t.end(), cmp);

    int cnt = solve(t);
    cout << cnt << endl;
    return 0;
}
